test/test_lib.py

```diff
@@ -145,6 +145,27 @@ def test_list_with_fqbn(run_command):
     assert data[0]["library"]["compatible_with"]["arduino:avr:uno"]
 
 
+def test_list_all_with_fqbn(run_command):
+    assert run_command("update")
+
+    # Install core
+    assert run_command("core install arduino:avr")
+
+    result = run_command("lib list --all --fqbn arduino:avr:uno --format json")
+    assert result.ok
+    assert "" == result.stderr
+
+    data = json.loads(result.stdout)
+    assert 5 == len(data)
+
+    libs = {l["library"]["name"]: l["library"]["provides_includes"] for l in data}
+    assert ["SoftwareSerial.h"] == libs["SoftwareSerial"]
+    assert ["Wire.h"] == libs["Wire"]
+    assert ["EEPROM.h"] == libs["EEPROM"]
+    assert ["HID.h"] == libs["HID"]
+    assert ["SPI.h"] == libs["SPI"]
```

Comment:
Just to be clear, his bug is not in any way specific to the `--all` flag. It occurrs with any library that didn't have a library.properties `includes` field, regardless of where that library was installed. The issue just happened to be encountered with platform bundled libraries.

So I don't think this is an ideal test for the bug fixed by this PR. If an `includes` field was added to the library.properties files of these libraries it would not be obvious that the test was no longer covering the new code. Better for that purpose would be a test (e.g., `test_list_provides_includes_fallback`) that uses a minimal test data library without an `includes` field. I suppose a balance needs to be struck between good coverage and a manageable number of integration tests though. Perhaps a dedicated test for this field alone is getting too fine-grained. I do note that this is the second bug we've had with the handling of the `includes` field though (https://github.com/arduino/arduino-cli/issues/960).

Reply:
True that, it's not exactly specific to only the `--all` flag, I tend to write tests for the specific issue I got reported, I'll make it a bit more generic by adding a library that doesn't have an `includes` property.

Since this is an integration test we don't really need to verify the code behavior, that's the job of unit tests. 
If the code that creates this output changes it's not a concern of this test, it has to just verify that the output is consistent and not the way it's generated. 

Nonetheless I'll change the test to install a specific version of a core and a library with the missing `includes` property, and it's name so that is more descriptive.